### ISSUE TITLE
Ability to cancel a query

### ISSUE THREAD

**Issue by sjmcdowall** [zino-app/graphql-flutter#181]
**Is your feature request related to a problem? Please describe.**

We have a "costly" query that, alas, the user is calling over and over before the first one is even completed causing resources etc. to be used (and a huge slowdown on the system).  If the user submits a 2nd (or 3rd) query we'd like to cancel the prior one and submit the new one .. 

**Describe the solution you'd like**

I believe Apollo has a .cancel() method on the Query object.  That seems like a good solution :) 

**Describe alternatives you've considered**
We already have a keyboard debounce in place (300ms) but that isn't helping enough, the user is still able to submit a ton of queries and tie up the system. 

**Additional context**
Can't think of anything.

**Comment by micimize** [comment]
You might be able to use [the socket client](https://github.com/zino-app/graphql-flutter/tree/next#subscriptions-experimental), as it [implements cancel](https://github.com/zino-app/graphql-flutter/blob/d108eeff59619bda9b8b8365f2d67cbbd5b64ce1/lib/src/socket_client.dart#L60).

As for `GQL_STOP` in the http client, support could take a while.

But if you can get it working with the socket subscription, an example would help our docs a lot 😄

**Comment by normancarcamo** [comment]
Hello there, is this already implemented?

**Comment by micimize** [comment]
@normancarcamo no, if it was implemented this issue would be closed.

**Comment by SalahAdDin** [comment]
We are looking something like [**dio**'s cancel token](https://pub.dev/documentation/dio/latest/dio/CancelToken-class.html), right?

**Comment by lekhactho111** [comment]
same issue any update for this stuff? How is possible to cancel the previous request before I create a new one to reduce the workload to the backend side?

**Comment by yeras-is** [comment]
@micimize  any updates  ?

**Comment by vincenzopalazzo** [comment]
>any updates ?

No, this is hide for all the issue of the repository, so we don't start to work on it. at list we can start to look into it now

**Comment by ezze** [comment]
> > any updates ?
> 
> No, this is hide for all the issue of the repository, so we don't start to work on it. at list we can start to look into it now

@vincenzopalazzo We are also very interested in this feature. Currently, implementing product search with GraphQL we have to ignore responses from all previous pending requests. It doesn't seem good due to performance reasons.

**Comment by krishna-pietech** [comment]
@vincenzopalazzo We also need this feature for our app. Please consider this in future releases.

**Comment by appinteractive** [comment]
Hey any updates on that one here?

**Comment by vincenzopalazzo** [comment]
We didi not discuss about it yet, but PR are always welcome

**Comment by EhabSalah** [comment]
Any update guys?

**Comment by simonvorac** [comment]
Hi, what's the latest status please or has anyone figured out how to solve this? 
We use graphql on Web and this is a big issue for us as switching to a different section waits for pending queries from previous one to finish first before loading. Switching quickly between multiple pages just hangs for a very long waiting time as it's waiting for all the pending queries to finish...

**Comment by appinteractive** [comment]
@zino-hofmann, @micimize, @HofmannZ, @vincenzopalazzo, @mainawycliffe, @truongsinh any news on this one?

**Comment by micimize** [comment]
@appinteractive I and the majority of the past contributors you just tagged have long since stopped maintaining this project.

If I recall correctly, this would be easier after a migration to [gql_dio_link](https://github.com/gql-dart/gql/tree/master/links/gql_dio_link). It seems fairly involved to implement though.

Anyone who really needs this should prob just implement it themselves – I think the [graphql/ARCHITECTURE.md](https://github.com/zino-hofmann/graphql-flutter/blob/main/packages/graphql/ARCHITECTURE.md) doc is still fairly accurate.

**Comment by appinteractive** [comment]
@micimize thanks for your response, that's sad to hear that multiple of the most active contributors stopped working on it. But do you know if it's still actively maintained, though? And thanks for the hints! 

Anyone skilled here to take a look on that maybe? 👀

**Comment by micimize** [comment]
@appinteractive yeah it is still actively maintained/updated, though infrequently. I don't keep track though

**Comment by Subhangi3** [comment]
Any update on this?

**Comment by wildsurfer** [comment]
Also facing the necessity of that feature.

**Comment by MostafaSolimanMO** [comment]
Wow, 5 years.

**Comment by nandaprasetio97** [comment]
Any update on this?

**Comment by SalahAdDin** [comment]
> Wow, 5 years.

Six years.

**Comment by HofmannZ** [comment]
@MostafaSolimanMO @nandaprasetio97 @SalahAdDin @wildsurfer - Feel fee to contribute with a PR :)

**Comment by vincenzopalazzo** [comment]
Please the people that are interested in this, review this PR https://github.com/zino-hofmann/graphql-flutter/pull/1518
